### ISSUE TITLE
Nullable Reference Types false warnings in IDE from project configuration

### ISSUE THREAD

**Issue by yaakov-h** [dotnet/roslyn#31852]
**Version Used**: 

Microsoft Visual Studio Enterprise 2019 Preview
Version 16.0.0 Preview 1.1

**Steps to Reproduce**:

1. Create a new `Class Library (.NET Framework)` project
2. Create `Directory.Build.props` in the parent directory with the following contents:

```xml
<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <!-- General build settings -->
    <PropertyGroup>
        <LangVersion>8.0</LangVersion>
        <NullableReferenceTypes>true</NullableReferenceTypes>
    </PropertyGroup>
</Project>
```

3. Replace the contents of `Class1.cs` with the following:

```csharp
namespace Foo
{
    public class Bar
    {
        public void Baz(object? o)
        {
        }
    }
}
```

4. Restart Visual Studio
5. Open the above solution
6. Rebuild the project

**Expected Behavior**:

- No design-time errors
- No design-time warnings
- No compile-time errors
- No compile-time warnings

**Actual Behavior**:

- Design-time errors, one or more of:
  - CS8107: Feature 'nullable reference types' is not available in C# 7.0. Please use language version 8.0 or greater.
  - CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.

**Comment by gafter** [comment]
Why are these errors unexpected?  This feature is only available in C# 8, so you need to specify that.  And the annotation on a reference type such as `object?` is only permitted after `#nullable enable`.  So both errors are correct.

**Comment by yaakov-h** [comment]
LangVersion is set to 8.0 though.

**Comment by gafter** [comment]
Can you please share the project file?

**Comment by yaakov-h** [comment]
```xml
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
    <ProjectGuid>258b25ab-50a1-459c-9626-7954989e72cf</ProjectGuid>
    <OutputType>Library</OutputType>
    <AppDesignerFolder>Properties</AppDesignerFolder>
    <RootNamespace>RoslynNullableWarning</RootNamespace>
    <AssemblyName>RoslynNullableWarning</AssemblyName>
    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
    <FileAlignment>512</FileAlignment>
    <Deterministic>true</Deterministic>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Debug\</OutputPath>
    <DefineConstants>DEBUG;TRACE</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
    <DebugType>pdbonly</DebugType>
    <Optimize>true</Optimize>
    <OutputPath>bin\Release\</OutputPath>
    <DefineConstants>TRACE</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
  </PropertyGroup>
  <ItemGroup>
    <Reference Include="System"/>
    
    <Reference Include="System.Core"/>
    <Reference Include="System.Xml.Linq"/>
    <Reference Include="System.Data.DataSetExtensions"/>
    
    
    <Reference Include="Microsoft.CSharp"/>
    
    <Reference Include="System.Data"/>
    
    <Reference Include="System.Net.Http"/>
    
    <Reference Include="System.Xml"/>
  </ItemGroup>
  <ItemGroup>
    <Compile Include="Class1.cs" />
    <Compile Include="Properties\AssemblyInfo.cs" />
  </ItemGroup>
  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
 </Project>
```

**Comment by ufcpp** [comment]
`Directory.Build.props ` can only work for SDK-based project?

**Comment by yaakov-h** [comment]
@ufcpp That is incorrect.

**Comment by oscarvarto** [comment]
I can confirm the same situation that  @yaakov-h is getting:

I am getting warnings for .NET Framework projects, but it works correctly with .NET Core 3 preview projects.

Example of some warnings that appear in .NET Framework project:

+ AD0001	Analyzer 'Microsoft.CodeAnalysis.Formatting.FormattingDiagnosticAnalyzer' threw an exception of type 'System.InvalidOperationException' with message 'Unexpected false'.
+ CS8632	The annotation for nullable reference types should only be used in code within a '#nullable' context.

**Comment by jeremymeng** [comment]
I have the same settings in `Directory.Build.props` and seeing CS8632 warnings for an ASP.NET Core 2.1 project

```xml
    <LangVersion>8.0</LangVersion>
    <NullableReferenceTypes>true</NullableReferenceTypes>
```

Repro shared at an internal drop location `\\scratch2\scratch\public\yumeng\ASP.NET Core Demo.zip`

**Comment by newbe36524** [comment]
option has changed. NullableReferenceTypes -> NullableContextOptions
refer to : https://github.com/dotnet/docs/issues/9517#issuecomment-457630516

**Comment by jcouv** [comment]
I have not observed this issue in 16.1 or 16.2, and we have received no other report.
I'll go ahead and close the issue. Please ping or re-open if you encounter this on a recent version. Thanks
